backend/src/main/kotlin/org/loculus/backend/service/submission/FileMappingPreconditionValidator.kt

```diff
@@ -75,8 +64,29 @@ class FileMappingPreconditionValidator(
         fileCategoriesFilesMap.categories.forEach { category: FileCategory ->
             if (!allowedCategoriesName.contains(category)) {
                 throw UnprocessableEntityException(
-                    "The category $category is not part of the configured categories for ${organism.name}. " +
-                        "Allowed categories are: ${allowedCategoriesName.joinToString(", ")}.",
+                    "The category $category is not part of the configured submission categories for ${organism.name}." +
+                        " Allowed categories are: ${allowedFileCategories.joinToString(", ")}.",
+                )
+            }
+        }
+        return this
+    }
+
+    fun validateCategoriesMatchOutputSchema(
```

Comment:
Move this back up to the original place of the `fun validateCategoriesMatchProcessedSchema(` to make the diff smaller and the PR easier to review